src/fcc_bridge_node_ros.cpp

```diff
@@ -0,0 +1,225 @@
+//
+// Created by Johan <job8197@thi.de> on 30.04.2024.
+//
+
+#include "fcc_bridge_node.hpp"
+
+// Libc header
+#include <chrono>
+
+namespace fcc_bridge {
+FCCBridgeNode::FCCBridgeNode(const std::string &name/*,
+                             const rclcpp::NodeOptions &node_options*/)
+    : CommonNode(name), internal_state(INTERNAL_STATE::STARTING_UP) {
+    // Pre-populate the last mission control heartbeat
+    this->last_mission_control_heartbeat.time_stamp = this->now();
+
+    // Setup ROS objects such as timer, publishers etc.
+    this->setup_ros();
+    if (this->internal_state == INTERNAL_STATE::ERROR) {
+        RCLCPP_FATAL(this->get_logger(), "Failed to setup ROS! Exiting...");
+        this->exit_process_on_error();
+    }
+    this->internal_state = INTERNAL_STATE::ROS_SET_UP;

```

Comment:
I suggest using a function to set a new state.
This way, you can trigger the log message in the function instead.

Reply:
Implemented in b831b9d